user-guide/modules/ROOT/pages/getting-started.adoc

```diff
@@ -0,0 +1,886 @@
+By supporting CMake, Visual Studio is able to provide a consistent development experience across multiple platforms and build tools.
+Additionally, the ability to generate CMake projects from Visual Studio allows developers to take advantage of Visual Studio's debugging and profiling tools while still using their preferred build system.
+
+While Visual Studio provides a user friendly environment for building software, it may not always be the best choice for managing complex projects with multiple dependencies.
+In such cases, it is recommended to consider using CMake, a popular and flexible build system that can generate build files for a variety of build tools.
+You can have the option of using CMake to manage your complex projects, while still using Visual Studio's robust development and debugging features.
+====
+
+--
+
+By Hand::
++
+--
+In the directory where you saved *example.cpp*, issue the following command:
+
+Visual Studio Compiler:
+
+[source,none,subs="attributes+"]
+----
+cl /I C:\boost\include example.cpp
+----
+
+GCC:
+
+[source,none,subs="attributes+"]
+----
+g++ -I /usr/local/include example.cpp -o example
+----
+
+Clang:
+
+[source,none,subs="attributes+"]
+----
+clang++ -I /usr/local/include example.cpp -o example
+----
+
+NOTE: Replace `C:\boost` or `/usr/local` with your Boost installation prefix if necessary.
+
+[TIP]
+====
+It is generally not recommended to use these commands by hand for complex projects, as managing all the compile and link options can quickly become unwieldy.
+Using a build system like CMake or Make can greatly simplify the process of building and managing a project, particularly when dealing with large codebases or dependencies.
+
+Using a build system also has other advantages, such as easier dependency management, the ability to easily configure the build for different platforms or compilers, and the ability to easily integrate with other tools such as version control systems and automated testing frameworks.
+====
+
+--
+========
+
+To test the result, run the executable from the path where it was built:
+
+[tabs,sync-group-id=os]
+====
+Windows::
++
+--
+[source]
+----
+echo 1 2 3 | example
+----
+--
+
+Linux::
++
+--
+[source]
+----
+echo 1 2 3 | ./example
+----
+--
+
+macOS::
++
+--
+[source]
+----
+echo 1 2 3 | ./example
+----
+--
+====
+
+Did you get the expected result?
+
+=== Compiled Libraries
+
+In this section, we will locate an example from GitHub, and then build and run a project based on the example.
+
+For most Boost libraries, there is an `example` sub-folder containing a range of examples.
+For this guide, we will locate an example that reads a JSON file, and pretty-prints its contents.
+
+Copy all the code from https://github.com/boostorg/json/blob/develop/example/pretty.cpp[`json/example/pretty.cpp`,window="_blank"] into your `example.cpp`.
```

Comment:
While this is nice, if you dumb-follow the instructions, this won't work, since the example includes `"file.hpp"`.

Reply:
Yes. Same as the other example. We have to look for a good one. And test it in all dumb-follow paths.